core/src/main/java/org/apache/accumulo/core/client/impl/Namespaces.java

```diff
@@ -100,50 +106,150 @@ private static ZooCache getZooCache(Instance instance) {
       if (namespace.equals(Tables.qualify(name).getFirst()))
         names.add(name);
     return names;
   }
 
+  /**
+   * Populate map passed in as the BiConsumer. key = ID, value = namespaceName
+   */
+  private static void populateMap(Instance instance, BiConsumer<String,String> biConsumer) {
+    final ZooCache zc = getZooCache(instance);
+    List<String> namespaceIds = zc.getChildren(ZooUtil.getRoot(instance) + Constants.ZNAMESPACES);
+    for (String id : namespaceIds) {
+      byte[] path = zc.get(ZooUtil.getRoot(instance) + Constants.ZNAMESPACES + "/" + id + Constants.ZNAMESPACE_NAME);
+      if (path != null) {
+        biConsumer.accept(id, new String(path, UTF_8));
+      }
+    }
+  }
+
+  /**
+   * Return sorted map with key = ID, value = namespaceName
+   */
+  public static SortedMap<Namespace.ID,String> getIdMap(Instance instance) {
+    SortedMap<Namespace.ID,String> idMap = new TreeMap<>();
+    populateMap(instance, (id, name) -> idMap.put(new Namespace.ID(id), name));
+    return idMap;
+  }
+
+  /**
+   * Return sorted map with key = namespaceName, value = ID
+   */
+  public static SortedMap<String,Namespace.ID> getNameMap(Instance instance) {
+    SortedMap<String,Namespace.ID> nameMap = new TreeMap<>();
+    populateMap(instance, (id, name) -> nameMap.put(name, new Namespace.ID(id)));
+    return nameMap;
+  }
+
+  /**
+   * Look for namespace ID in ZK. Throw NamespaceNotFoundException if not found.
+   */
+  public static Namespace.ID getNamespaceId(Instance instance, String namespaceName) throws NamespaceNotFoundException {
+    final ArrayList<Namespace.ID> singleId = new ArrayList<>(1);
+    populateMap(instance, (id, name) -> {
+      if (name.equals(namespaceName))
+        singleId.add(new Namespace.ID(id));
+    });
+    if (singleId.isEmpty())
+      throw new NamespaceNotFoundException(null, namespaceName, "getNamespaceId() failed to find namespace");
+    return singleId.get(0);
+  }
+
+  /**
+   * Look for namespace ID in ZK. Fail quietly by logging and returning null.
+   */
+  public static Namespace.ID lookupNamespaceId(Instance instance, String namespaceName) {
+    Namespace.ID id = null;
+    try {
+      id = getNamespaceId(instance, namespaceName);
+    } catch (NamespaceNotFoundException e) {
+      if (log.isDebugEnabled())
+        log.debug("Failed to find namespace ID from name: " + namespaceName, e);
+    }
+    return id;
+  }
+
+  /**
+   * Return true if namespace name exists
+   */
+  public static boolean namespaceNameExists(Instance instance, String namespaceName) {
+    return lookupNamespaceId(instance, namespaceName) != null;
+  }
+
+  /**
+   * Look for namespace name in ZK. Throw NamespaceNotFoundException if not found.
+   */
+  public static String getNamespaceName(Instance instance, Namespace.ID namespaceId) throws NamespaceNotFoundException {
+    final ArrayList<String> singleName = new ArrayList<>(1);
+    populateMap(instance, (id, name) -> {
```

Comment:
This is similar to what we had:
`String namespaceName = getIdToNameMap(instance).get(namespaceId);`

But uses the new populateMap method I created.  I created the list so we could pass a mutable object to the BiConsumer.  I didn't think a String would work since they are immutable.  I could change this method back to operate the old way, if you think that is cleaner.  It seems like some situations Java 8 code looks messier.